app/src/main/java/com/codingblocks/chatter/fragments/RoomsFragment.java

```diff
@@ -288,4 +303,175 @@ protected void onPostExecute(List<RoomsTable> notes) {
+                getActivity().startActivity(intent);
+                getActivity().finish();
+            }
+            Request request = new Request.Builder()
+                    .url("https://api.gitter.im/v1/rooms?q=" + newText)
+                    .build();
+            HttpUrl url = request.url().newBuilder()
+                    .addQueryParameter("access_token", accessToken)
+                    .build();
+            request = request.newBuilder().url(url).build();
+            client.newCall(request).enqueue(new Callback() {
+                @Override
+                public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                    e.printStackTrace();
+                }
+
+                @Override
+                public void onResponse(@NonNull Call call, @NonNull Response response)
+                        throws IOException {
+                    /* Simple hack for compatibility as API 19 is required for
+                       new JSONArray */
+                    final String responseText = response.body().string();
+                    // We will move to UI Thread
+                    Thread thread = new Thread(new Runnable() {
+                        @SuppressLint("StaticFieldLeak")
+                        @Override
+                        public void run() {
+                            try {
+                                JSONObject JObject = new JSONObject(responseText);
+                                Log.i(TAG, "run: " + responseText);
+                                JSONArray JArray = JObject.getJSONArray("results");
+                                int i;
+                                for (i = 0; i < JArray.length(); i++) {
+                                    JSONObject dynamicJObject = JArray.getJSONObject(i);
+                                    String githubType = dynamicJObject.getString("githubType");
+                                    final String uId = dynamicJObject.getString("id");
+                                    String name = dynamicJObject.getString("name");
+                                    String url = dynamicJObject.getString("avatarUrl");
+                                    // userCount = 0 == user to user room since ONETWOONE doesnot have it
+                                    int userCount = 0;
+                                    if (!githubType.equals("ONETWOONE")) {
+                                        userCount = dynamicJObject.getInt("userCount");
+                                    }
+                                    int unreadItems = dynamicJObject.getInt("unreadItems");
+                                    int mentions = dynamicJObject.getInt("mentions");
+                                    boolean roomMember = dynamicJObject.getBoolean("roomMember");
+                                    Log.i(TAG, "run: " + dynamicJObject.toString());
+                                    final RoomsTable room = new RoomsTable();
```

Comment:
Ain't this logically incorrect? Why do you add new rooms to the db just by searching? The rooms are added to the db if a user has joined them. This will result in unjoined rooms being pushed in the dashboard.

Reply:
i'm not adding the room to db just adding to the list check again